components/createCollection/AddNftStep.tsx

```diff
@@ -0,0 +1,255 @@
+    initialState,
+  );
+  const [formDataErrors, setFormDataErrors] = useState<
+    StepTwoErrors | NftFormErrors
+  >(initialErrors);
+  const handleFormChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.currentTarget.type === 'checkbox') {
+      setNftFormData({
+        ...nftFormData,
+        [event.currentTarget.name]: event.currentTarget.checked,
+      });
+    } else {
+      // form Validation for events.
+      if (event.target.value.length === 0) {
+        setFormDataErrors({
+          ...formDataErrors,
+          [`${event.target.name}Error`]: true,
+        });
+      } else {
+        setFormDataErrors({
+          ...formDataErrors,
+          [`${event.target.name}Error`]: false,
+        });
+      }
+      setNftFormData({
+        ...nftFormData,
+        [event.currentTarget.name]: event.currentTarget.value,
+      });
+    }
+  };
+  const checkFormValidated = () => {
+    const validated = Object.values(formDataErrors).every(
+      (item) => item === false,
+    );
+    return validated;
+  };
+  const deleteNft = (tokenName: string) => {
+    const updatedArray = formData.nfts.filter(
+      (item) => item.tokenName !== tokenName,
+    );
+    setFormData({
+      ...formData,
+      nfts: updatedArray,
+    });
+  };
+  const editNftItem = (nft: NftInCollection) => {
+    setEditNft(true);
+    setNftFormData(nft);
+    setAddNft(true);
+    const updatedArray = formData.nfts.filter(
+      (item) => item.tokenName !== nft.tokenName,
+    );
+    setFormData({
+      ...formData,
+      nfts: updatedArray,
+    });
+  };
+
+  const addNftConfirm = () => {
+    setAddNft(false);
+    setFormData((prev) => ({
+      ...prev,
+      nfts: [...prev.nfts, nftFormData as NftInCollection],
+    }));
+  };
+  const editNftConfirm = () => {
+    setAddNft(false);
+    setEditNft(false);
+    setFormData((prev) => ({
+      ...prev,
+      nfts: [...prev.nfts, nftFormData as NftInCollection],
+    }));
+  };
+
+  const addNftInCollection = () => {
+    setNftFormData(initialState);
+    setFormDataErrors(initialErrors);
+    setAddNft(true);
+  };
+  if (!addNft) {
+    return (
+      <>
+        <h1 className="text-3xl mt-16 font-bold">Create your NFTs</h1>
+        <h4 className="text-sm mt-2">Basic Collection Information</h4>
+
+        <div className="py-7 border-b grid grid-cols-2 md:grid-cols-3 gap-2">
+          <div role="presentation" onClick={addNftInCollection}>
+            <NFTCard emptyCard />
+          </div>
+          {formData.nfts.map((nft) => (
+            <NFTCard
+              deleteNft={() => deleteNft(nft.tokenName)}
+              editNft={() => editNftItem(nft)}
+              image={URL.createObjectURL(nft.nftThumbnail!)}
+              nftName={nft.tokenName}
+              collectionName={nft.creatorName}
+            />
+          ))}
+        </div>
+        <Button
+          onClick={() => {
+            setOpenConfirmMintModal(true);
```

Comment:
Move confirm modal to "Verify and Mint" Screen. It should open after clicking "Mint now".

Reply:
done